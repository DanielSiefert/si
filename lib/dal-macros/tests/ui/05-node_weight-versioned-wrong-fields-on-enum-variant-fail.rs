pub struct InnerType;

pub trait SiVersionedNodeWeight {
    type Inner;

    fn inner(&self) -> &Self::Inner;
    fn inner_mut(&mut self) -> &mut Self::Inner;
}

#[derive(dal_macros::SiVersionedNodeWeight)]
pub enum VersionedNodeWeight {
    #[si_versioned_node_weight(current)]
    V1(InnerType),
    #[si_versioned_node_weight(current)]
    V2(InnerType, InnerType),
    #[si_versioned_node_weight(current)]
    V3 {
        inner: InnerType,
    },
    #[si_versioned_node_weight(current)]
    V4,
    V5,
    V6(InnerType),
    V7(InnerType, InnerType),
    V8 {
        inner: InnerType,
    },
}

fn main() {}
