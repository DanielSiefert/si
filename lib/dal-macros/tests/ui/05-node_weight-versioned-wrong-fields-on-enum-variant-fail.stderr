error: Enum cannot have multiple current variants.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:14:5
   |
14 | /     #[si_versioned_node_weight(current)]
15 | |     V2(InnerType, InnerType),
   | |____________________________^

error: Current variant must have a single unnamed field.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:15:7
   |
15 |     V2(InnerType, InnerType),
   |       ^^^^^^^^^^^^^^^^^^^^^^

error: Enum cannot have multiple current variants.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:16:5
   |
16 | /     #[si_versioned_node_weight(current)]
17 | |     V3 {
18 | |         inner: InnerType,
19 | |     },
   | |_____^

error: Current variant must have a single unnamed field.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:17:8
   |
17 |       V3 {
   |  ________^
18 | |         inner: InnerType,
19 | |     },
   | |_____^

error: Enum cannot have multiple current variants.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:20:5
   |
20 | /     #[si_versioned_node_weight(current)]
21 | |     V4,
   | |______^

error: Current variant must have a single unnamed field.
  --> tests/ui/05-node_weight-versioned-wrong-fields-on-enum-variant-fail.rs:20:5
   |
20 | /     #[si_versioned_node_weight(current)]
21 | |     V4,
   | |______^
